### PR TITLE
fix: announcements weekly count display in challenges overview - MEED-1414 - Meeds-io/MIPs#10 

### PR DESCRIPTION
**Pull request by ArfaouiMohamedAmine**
this change is going to fix the count of the announecements displayed in challenges overview

### CODE REVIEW

File: portlets/src/main/webapp/vue-app/challengesOverview/components/ChallengesOverview.vue [hunk 1]
```diff
@@ -95,8 +95,9 @@ export default {
   data: () => ({
     search: '',
     challengePerPage: 3,
-    announcementsPerChallenge: 2,
+    announcementsPerChallenge: -1,
     filter: 'STARTED',
+    dateBornsFilter: 'WEEK',
```
Comment (AzmiTouil) [inline portlets/src/main/webapp/vue-app/challengesOverview/components/ChallengesOverview.vue]:
```suggestion
    period: 'WEEK',
```

File: portlets/src/main/webapp/vue-app/engagement-center/js/challengesServices.js [hunk 2]
```diff
@@ -65,7 +65,7 @@ export function updateChallenge(challenge) {
   });
 }
 
-export function getAllChallengesByUser(term , offset, limit, announcements, domainId, groupByDomain, filter, orderByRealizations, excludedChallengesIds) {
+export function getAllChallengesByUser(term , offset, limit, announcements, domainId, groupByDomain, filter, orderByRealizations, excludedChallengesIds, dateBornsFilter) {
```
Comment (AzmiTouil) [inline portlets/src/main/webapp/vue-app/engagement-center/js/challengesServices.js]:
```suggestion
export function getAllChallengesByUser(term , offset, limit, announcements, domainId, groupByDomain, filter, orderByRealizations, excludedChallengesIds, period) {
```

File: portlets/src/main/webapp/vue-app/engagement-center/js/challengesServices.js [hunk 3]
```diff
@@ -96,6 +96,9 @@ export function getAllChallengesByUser(term , offset, limit, announcements, doma
       formData.append('excludedChallengesIds', element);
     }
   }
+  if (dateBornsFilter) {
+    formData.append('dateBornsFilter', dateBornsFilter);
```
Comment (AzmiTouil) [inline portlets/src/main/webapp/vue-app/engagement-center/js/challengesServices.js]:
```suggestion
    formData.append('period', period);
```

File: services/src/main/java/org/exoplatform/addons/gamification/service/dto/configuration/constant/DateBornsType.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+package org.exoplatform.addons.gamification.service.dto.configuration.constant;
+
+public enum DateBornsType {
```
Comment (AzmiTouil) [inline services/src/main/java/org/exoplatform/addons/gamification/service/dto/configuration/constant/DateBornsType.java]:
```suggestion
public enum PeriodType {
```

File: services/src/main/java/org/exoplatform/addons/gamification/storage/dao/GamificationHistoryDAO.java [hunk 5]
```diff
@@ -408,13 +410,31 @@ public Long countAnnouncementsByChallenge(Long challengeId) {
...
   }
 
-  public List<GamificationActionsHistory> findAllAnnouncementByChallenge(Long challengeId, int offset, int limit) {
-    TypedQuery<GamificationActionsHistory> query =
-                                                 getEntityManager().createNamedQuery("GamificationActionsHistory.findAllAnnouncementByChallenge",
-                                                                                     GamificationActionsHistory.class);
+  public List<GamificationActionsHistory> findAllAnnouncementByChallenge(Long challengeId,
+                                                                         int offset,
+                                                                         int limit,
+                                                                         RuleFilter ruleFilter) {
```
Comment (AzmiTouil) [inline services/src/main/java/org/exoplatform/addons/gamification/storage/dao/GamificationHistoryDAO.java]: no need to add the whole rule filter, just pass the period as a parameter.

**Review by AzmiTouil — CHANGES_REQUESTED** [review]

File: services/src/main/java/org/exoplatform/addons/gamification/storage/dao/GamificationHistoryDAO.java (hunk 5 above)
Comment (ArfaouiMohamedAmine) [reply services/src/main/java/org/exoplatform/addons/gamification/storage/dao/GamificationHistoryDAO.java]: okay am on it

File: services/src/main/java/org/exoplatform/addons/gamification/service/dto/configuration/constant/DateBornsType.java (hunk 4 above)
Comment (ArfaouiMohamedAmine) [reply services/src/main/java/org/exoplatform/addons/gamification/service/dto/configuration/constant/DateBornsType.java]: okay am on it

File: portlets/src/main/webapp/vue-app/engagement-center/js/challengesServices.js (hunk 3 above)
Comment (ArfaouiMohamedAmine) [reply portlets/src/main/webapp/vue-app/engagement-center/js/challengesServices.js]: okay am on it

File: portlets/src/main/webapp/vue-app/engagement-center/js/challengesServices.js (hunk 2 above)
Comment (ArfaouiMohamedAmine) [reply portlets/src/main/webapp/vue-app/engagement-center/js/challengesServices.js]: okay am on it

File: portlets/src/main/webapp/vue-app/challengesOverview/components/ChallengesOverview.vue (hunk 1 above)
Comment (ArfaouiMohamedAmine) [reply portlets/src/main/webapp/vue-app/challengesOverview/components/ChallengesOverview.vue]: okay am on it

**Comment by exo-swf** [comment]
Your PR triggers too many exo-ci builds! Please finish your work and then, set your PR ready! Thank you

**Review by AzmiTouil — APPROVED** [review]

**Pull request merged**